libc/test/src/math/smoke/FMinimumNumTest.h

```diff
@@ -0,0 +1,100 @@
+//===-- Utility class to test fminimum_num[f|l] -----------------*- C++ -*-===//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+
+#ifndef LLVM_LIBC_TEST_SRC_MATH_SMOKE_FMINIMUMNUMTEST_H
+#define LLVM_LIBC_TEST_SRC_MATH_SMOKE_FMINIMUMNUMTEST_H
+
+#include "src/__support/FPUtil/FPBits.h"
+#include "test/UnitTest/FPMatcher.h"
+#include "test/UnitTest/Test.h"
+
+template <typename T>
+class FMinimumNumTest : public LIBC_NAMESPACE::testing::Test {
+
+  DECLARE_SPECIAL_CONSTANTS(T)
+
+public:
+  typedef T (*FMinimumNumFunc)(T, T);
+
+  void testNaN(FMinimumNumFunc func) {
+    EXPECT_FP_EQ(inf, func(FPBits::quiet_nan().get_val(), inf));
+    EXPECT_FP_EQ(inf, func(FPBits::signaling_nan().get_val(), inf));
+    EXPECT_FP_EQ(neg_inf, func(neg_inf, FPBits::quiet_nan().get_val()));
+    EXPECT_FP_EQ(neg_inf, func(neg_inf, FPBits::signaling_nan().get_val()));
+    EXPECT_FP_EQ(FPBits::quiet_nan().get_val(), func(aNaN, aNaN));
+    EXPECT_FP_EQ(0.0, func(FPBits::quiet_nan().get_val(), 0.0));
+    EXPECT_FP_EQ(-0.0, func(-0.0, FPBits::quiet_nan().get_val()));
+    EXPECT_FP_EQ(0.0, func(FPBits::signaling_nan().get_val(), 0.0));
+    EXPECT_FP_EQ(-0.0, func(-0.0, FPBits::signaling_nan().get_val()));
+    EXPECT_FP_EQ(T(-1.2345), func(FPBits::quiet_nan().get_val(), T(-1.2345)));
+    EXPECT_FP_EQ(T(1.2345), func(T(1.2345), FPBits::quiet_nan().get_val()));
+    EXPECT_FP_EQ(T(-1.2345),
```

Comment:
Change this to `EXPECT_FP_EQ_WITH_EXCEPTION( ... , FE_INVALID);`

Reply:
You’re highlighting 4 lines. Which do I change? Do I change all the lines that you highlighted?

Reply:
Sorry, I meant that change all the `EXPECT_FP_EQ` to `EXPECT_FP_EQ_WITH_EXCEPTION` if one of the input is signaling nan.

Reply:
So do I need to change the logic in BasicOperations? Because in my code I only raise an exception when one argument is a nan while the other is a signaling nan

Reply:
Yes, section F.10.9.5 from https://www.open-std.org/jtc1/sc22/wg14/www/docs/n3096.pdf states that:
- If one input is a number, and the other input is a NaN (quiet or signaling), then return the number.
- If both inputs are NaN (quiet or signaling), then return a quiet NaN.
- If any of the inputs is a signaling NaN, the invalid floating point exception is raised (even if the other input is a number)

Reply:
> Yes, section F.10.9.5 from https://www.open-std.org/jtc1/sc22/wg14/www/docs/n3096.pdf states that:
> 
>     * If one input is a number, and the other input is a NaN (quiet or signaling), then return the number.
> 
>     * If both inputs are NaN (quiet or signaling), then return a quiet NaN.
> 
>     * If any of the inputs is a signaling NaN, the invalid floating point exception is raised (even if the other input is a number)
> 
> 
> So for updating the logic of `_num` functions, it might be easier to treat the case when neither of the inputs are NaNs first. Then you can deal with all the NaN cases afterward.
> 
> And `FPBits::is_nan()` function will return true for both quiet and signaling NaNs.

ok Ill try to address all of your review  in the next two hours.

Reply:
Just take your time, don't need to rush.  And thanks for bearing with me all the time!

Reply:
No problem. I just finished addressing your review! I’m looking forward to merge this! Thanks